torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -279,6 +279,8 @@ class TORCH_CUDA_API GridReduction : public Expr {
   Allocate* sync_buffer_ = nullptr;
 };
 
+std::string getPredicateFlagName(const TensorView* val);
```

Comment:
nice. wouldn't this fit better in `lower_thread_predicate.h` though?

I'd like to keep the IR as lightweight as possible (just a data structure). This means keeping lowering and the IR as decoupled as possible.

Reply:
I wasn't quite sure where this should be placed. I just picked `kernel_ir.h` just because it's related to the kernel IR. 

The function is also used from `IRPrinter`. I feel a little uneasy if I needed to include `lower_thread_predicate.h` from `ir_ioprinter.cpp`, but that might make more sense.